### ISSUE TITLE
Request to /service-worker.js responds with 404

### ISSUE THREAD

**Issue by JensMeindertsma** [hanford/next-offline#212]
**Repo:** _https://github.com/JensMeindertsma/next-offline-404_

I am trying to get `next-offline@5.0.0-beta.11` to work with `next@9.1.6`. I have created a basic project by running `npx create-next-app pwa` followed by `npm i next-offline@5.0.0-beta.11`. I have included this repository. I have created a basic `next.config.js` file that looks as follows:
```
const withOffline = require('next-offline')

const nextConfig = {};

module.exports = withOffline(nextConfig)
```

My server.js looks as follows ( _copy & paste from docs_ ):
```
const { createServer } = require('http')
const { join } = require('path')
const { parse } = require('url')
const next = require('next')

const app = next({ dev: process.env.NODE_ENV !== 'production' })
const handle = app.getRequestHandler()

app.prepare()
  .then(() => {
    createServer((req, res) => {
      const parsedUrl = parse(req.url, true)
      const { pathname } = parsedUrl

      // handle GET request to /service-worker.js
      if (pathname === '/service-worker.js') {
        const filePath = join(__dirname, '.next', pathname)

        app.serveStatic(req, res, filePath)
      } else {
        handle(req, res, parsedUrl)
      }
    })
      .listen(3000, () => {
        console.log(`> Ready on http://localhost:${3000}`)
      })
  })
```
**Steps to reproduce ( _from repo_ )**
When I run `npm run build` followed by `npm run start`, and browse to `http://localhost:3000/`, I see the default Next.js page. More importantly, the Chrome ( _v78.0.3904.108_ ) DevTools console has logged an error message:
`A bad HTTP response code (404) was received when fetching the script.` 
This is followed by:
`SW registration failed:  TypeError: Failed to register a ServiceWorker for scope ('http://localhost:3000/') with script ('http://localhost:3000/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.`

After reading a lot of issues about this same error message, I checked whether the `service-worker.js` file did exist inside the `.next` build folder. It is located in the root of that folder.
When I navigate Chrome to `localhost:3000/service-worker.js`, I see the 404 page from Next.js. I believe this confirms the problem lies with the following code inside `server.js`:
```
if (pathname === '/service-worker.js') {
        const filePath = join(__dirname, '.next', pathname)

        app.serveStatic(req, res, filePath)
      } else {
        handle(req, res, parsedUrl)
      }
```
I believe there is a issue with `next-offline` that causes it to not serve the service worker when a request is made to `/service-worker.js` but instead return a 404.

**Please help me solve this problem**

I thank you in advance!

**Comment by aalibaabaa** [comment]
Hi @JensMeindertsma ,

I had a similar issue. In my case, I set the distDir value to _next 

`const nextConfig = {distDir: '_next'};`

but forgot to change the path in the `server.js`

```
// old
const filePath = join(__dirname, '.next', pathname)

// new
const filePath = join(__dirname, '_next', pathname)
```

After changing this, the service-worker could be served again. 
Maybe you have a similar issue.

**Comment by JensMeindertsma** [comment]
Dear Alibaabaa, as you can see in the repo, everything is default with no settings changed, so this can't be the issue. Thanks for helping out though!

**Comment by JensMeindertsma** [comment]
@hanford could you please take a look at this issue? I've provided a repo and all steps to replicate

**Comment by JensMeindertsma** [comment]
**Solution**
After reading the docs at _[https://github.com/hanford/next-offline](url)_ again, and still not finding a solution, I checked the Next.js docs. Specifically, I looked at the section at _[https://github.com/zeit/next.js#custom-server-and-routing](url)_. A snippet from this section:

> When using a custom server with a server file, for example called server.js, make sure you update the scripts key in package.json to:
> 
> ```
> {
>   "scripts": {
>     "dev": "node server.js",
>     "build": "next build",
>     "start": "NODE_ENV=production node server.js"
>   }
> }
> ```

**When reading the _`next-offline`_ documentation again, this _very crucial_ information is not written about anywhere!** I do not believe this can be considered general knowledge, because it is not specified in the docs and any reader might think that by just creating a `server.js` file this custom configuration is used.

When I tried running the `npm run start` command after changing this "scripts" property in `package.json,` I got another error:
`“NODE_ENV” is not recognized as an internal or external command, operable command or batch file`

This Windows related problem is discussed here: _[https://stackoverflow.com/questions/11928013/node-env-is-not-recognized-as-an-internal-or-external-command-operable-comman](url)_.

My solution for this problem is to install `cross-env` ( _[https://www.npmjs.com/package/cross-env](url)_ ). This must be installed under devDependencies:
`npm install --save-dev cross-env`
The "scripts" property in `package.json` can then be changed to:
```
"scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "cross-env NODE_ENV=production node server.js"
  }
```
_( change `"dev": "next dev"` to `"dev": "node server.js"` if you want to also use this custom `server.js` configuration during development. The `cross-env` solution is not necessary here, since we're not specifying "NODE_ENV"  )._

The service worker now registers properly!

I would suggest @hanford to add a section to the docs that specifies this change in `package.json` is required to get `next-offline` working. A lot of beginners might come across this problem!
